File: 48. Rotate Image.java
class Solution {
    public void rotate(int[][] m) {
        int size = m.length -1;
        for (int level=0; level<m.length/2;level++) {
            for(int i = level; i<size-level; i++) {
                int temp = m[level][i];
                m[level][i] = m[size-i][level];
                m[size-i][level] = m[size-level][size-i];
                m[size-level][size-i] = m[i][size-level]; 
                m[i][size-level] = temp;
            }
        }
            
    }
}

File: python/48. Rotate Image.py
# 48. Rotate Image
# https://leetcode.com/problems/rotate-image/

class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        size = len(matrix) -1
        for level in range((size + 1)//2):
            for x in range(level, size - level):
                temp = matrix[x][size-level]
                matrix[x][size-level] = matrix[level][x]
                matrix[level][x] = matrix[size - x][level]
                matrix[size - x][level] = matrix[size - level][size - x]
                matrix[size - level][size - x] = temp
                           
        
